### PR TITLE
[Relay] [WIP] Lambda Lifting

### PR DESCRIPTION
**Pull request by MarisaKirisame**
Jared need some code so I will make a pr right now

### CODE REVIEW

File: include/tvm/relay/pass.h [hunk 1]
```diff
@@ -92,6 +92,20 @@ bool AlphaEqual(const Type& t1, const Type& t2);
  */
 bool WellFormed(const Expr & e);
 
+/*! \brief Transform a program, such that every definition is, directly nested lambda, with the inner body having no more lambda.
+ *
+ * \example (\x -> \y -> z) is fine, but (\x -> map (\y -> z)) is not.
+ *
+ * It is achieved by generating a global definition for every lambda, with bounded variables transformed into function parameters.
```
Comment (jroesch) [inline include/tvm/relay/pass.h]: bound* variables.

**Comment by jroesch** [comment]
I think we should think about naming since this is a strange inbetween of classical lambda lifting and closure conversion.

**Comment by MarisaKirisame** [comment]
I can remove the closure conversion pass. It is just that I am clueless on the case of 'function returning closure'.

**Comment by yzhliu** [comment]
@MarisaKirisame Is it still WIP? or ready for merging?

**Comment by MarisaKirisame** [comment]
@yzhliu it was on hold, as we are working on more important things.

**Comment by tqchen** [comment]
close this for now as it is on hold, please reopen when there are updates

**Pull request closed without merging**